test/preview/dataclasses/test_document.py

```diff
@@ -224,117 +204,7 @@ def __eq__(self, other):
         dataframe=pd.DataFrame([10, 20, 30]),
         blob=None,
         mime_type="application/pdf",
-        # Note the object serialization
-        metadata={"some object": "<the object>", "a path": str((tmp_path / "test.txt").absolute())},
+        metadata={"create_date": -14186580},
         score=0.5,
         embedding=[1, 2, 3, 4],
     )
-
-
-@pytest.mark.unit
-def test_to_json_custom_encoder():
-    class SerializableTestClass:
-        ...
-
-    class TestEncoder(DocumentEncoder):
-        def default(self, obj):
-            if isinstance(obj, SerializableTestClass):
-                return "<<CUSTOM ENCODING>>"
-            return DocumentEncoder.default(self, obj)
-
-    doc = Document(text="test text", metadata={"some object": SerializableTestClass()})
-    doc_json = doc.to_json(indent=4, json_encoder=TestEncoder).strip()
-
-    assert doc_json == json.dumps(
-        {
-            "id": doc.id,
-            "text": "test text",
-            "dataframe": None,
-            "mime_type": "text/plain",
-            "metadata": {"some object": "<<CUSTOM ENCODING>>"},
-            "score": None,
-            "embedding": None,
-        },
-        indent=4,
-    )
-
-
-@pytest.mark.unit
-def test_from_json_custom_decoder():
-    class TestClass:
-        def __eq__(self, other):
-            return type(self) == type(other)
-
-    class TestDecoder(DocumentDecoder):
-        def __init__(self, *args, **kwargs):
-            super().__init__(object_hook=self.object_hook)
-
-        def object_hook(self, dictionary):
-            if "metadata" in dictionary:
-                for key, value in dictionary["metadata"].items():
-                    if value == "<<CUSTOM ENCODING>>":
-                        dictionary["metadata"][key] = TestClass()
-            return dictionary
-
-    doc = Document(text="test text", metadata={"some object": TestClass()})
-
-    assert doc == Document.from_json(
-        json.dumps(
-            {
-                "id": doc.id,
-                "text": "test text",
-                "dataframe": None,
-                "mime_type": "text/plain",
-                "metadata": {"some object": "<<CUSTOM ENCODING>>"},
-                "score": None,
-                "embedding": None,
-            }
-        ),
-        json_decoder=TestDecoder,
-    )
-
-
-@pytest.mark.unit
-def test_flatten_document_no_meta():
-    doc = Document(text="test text")
-    assert doc.flatten() == {
-        "id": doc.id,
-        "text": "test text",
-        "dataframe": None,
-        "blob": None,
-        "mime_type": "text/plain",
-        "score": None,
-        "embedding": None,
-    }
-
-
-@pytest.mark.unit
-def test_flatten_document_with_flat_meta():
-    doc = Document(text="test text", metadata={"some-key": "a value", "another-key": "another value!"})
```

Comment:
do we have tests for Document init with and without custom parameters? I can't find them right now.